packages/amplify_api_rest/test/amplify_api_rest_methods_test.dart

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:typed_data';
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_api_rest/amplify_api_rest.dart';
+
+void main() {
+  const MethodChannel apiRestChannel =
+      MethodChannel('com.amazonaws.amplify/amplify_api_rest');
+
+  const MethodChannel coreChannel = MethodChannel('com.amazonaws.amplify/core');
+
+  AmplifyAPIRest apiRest = new AmplifyAPIRest();
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  test('PUT returns proper response.data', () async {
+    var responseData = Uint8List.fromList(
+        "{\"success\": \"put call succeed!\",\"url\":\"/items?queryParameterA=queryValueA&queryParameterB=queryValueB\",\"body\": {\"name\": \"Mow the lawn\"}}"
+            .codeUnits);
+
+    apiRestChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == "put") {
+        return {"data": responseData};
+      }
+    });
```

Comment:
we should put an expectation on the input arguments (`RestOptions`) to be correctly serialized.

Reply:
updated all dart unit tests to check rest option values in merged branch.